hnsendpoint.go

```diff
@@ -192,18 +192,22 @@ func (endpoint *HNSEndpoint) ContainerHotDetach(containerID string) error {
 	return modifyNetworkEndpoint(containerID, endpoint.Id, Remove)
 }
 
-// ApplyACLPolicy applies Acl Policy on the Endpoint
-func (endpoint *HNSEndpoint) ApplyACLPolicy(policy *ACLPolicy) error {
+// ApplyPolicy applies a set of ACL Policies on the Endpoint. This will
+// overwrite any existing policies which were previously applied
+func (endpoint *HNSEndpoint) ApplyACLPolicy(policies []*ACLPolicy) error {
 	operation := "ApplyACLPolicy"
 	title := "HCSShim::HNSEndpoint::" + operation
 	logrus.Debugf(title+" id=%s", endpoint.Id)
 
-	jsonString, err := json.Marshal(policy)
-	if err != nil {
-		return err
+	for _, policy := range policies {
+		jsonString, err := json.Marshal(policy)

```

Comment:
Does HNS handle the case of a list of ACL policies with a nil entry in the middle of the list? Maybe we should just skip if policy is nil.

Reply:
HNS would be ok, but golang will be unhappy about this (calling Marshal on a nil will lead to a panic). Added a check for it.